solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -624,7 +628,15 @@ IndexFetchResult fetchLatestIndex(boolean forceReplication, boolean forceCoreRel
               .getDirectoryFactory()
               .get(indexDirPath, DirContext.DEFAULT, solrCore.getSolrConfig().indexConfig.lockType);
 
-      try {
+      Lock lock = tmpIndexDir.obtainLock(IndexWriter.WRITE_LOCK_NAME);
```

Comment:
not sure about this change as well?

Reply:
We only want to run TeeDirectory on the index dir, but there are other directories (that should just do normal file stuff) that are tracked through here. The reliable, if a bit ugly, way to detect this is to ensure that every directory that is being used for index purposes will have `obtainLock()` called on it. This always happens before the directory is read from or written to, except for in a few places where it was not strictly necessary because the directory reference is only held by local code (here and in `RestoreCore`).